### DISCUSSION TITLE
点进除了主页外的所有页面都会报错Cannot read properties of undefined (reading 'get')

### DISCUSSION THREAD

**Discussion by ERROR403XD** [Tokinx/Wing#105] [问答]
一开始我是用Adams的……然后出了一个bug，评论发不出去
然后我就摸到了Wing，然后发现点进每篇文章都会报Cannot read properties of undefined (reading 'get')这个错误。而且看起来报不止一次，像是加载什么资源的时候就会报这个错。
之前有的评论也加载不出来，尝试加载也会报这个错。
插件已经全都禁用掉了……

**Comment by Tokinx** [comment]
留个url？

**Reply by ERROR403XD** [reply]
> 留个url？

非常抱歉，因为我刚建站，所以就趁着东西还不多把主题换掉了……
我可以提供一些信息，我的WP版本是最新的（6.1.1），PHP版本是7.4，MySQL版本是5.6，系统版本是centos8
这个问题可以稳定复现，包括删站后裸装Wings也会出现。
一开始换Wings是因为Adams谜之发不出评论（换成WP自带的2021等主题可以正常评论，在Adams上，PC段不报错，移动端报undefined）。换成Wings后变成了上述错误。我尝试过关闭PJAX、删除所有插件、清空缓存等，都没有效果。
但有一个方法我没有尝试过，就是仅把站点地址改成https（在部署了SSL的前提下），这个改动修复了一些其他插件的问题（包括块编辑器不停掉线无法更新，一些插件无法安装license等），这是我在更换主题之后才发现的。
不知道上述信息对你有没有帮助，如果必要的话我想我可以再建一个站试试。

**Reply by moxiaonai** [reply]
> > 留个url？
> 
> 非常抱歉，因为我刚建站，所以就趁着东西还不多把主题换掉了…… 我可以提供一些信息，我的WP版本是最新的（6.1.1），PHP版本是7.4，MySQL版本是5.6，系统版本是centos8 这个问题可以稳定复现，包括删站后裸装Wings也会出现。 一开始换Wings是因为Adams谜之发不出评论（换成WP自带的2021等主题可以正常评论，在Adams上，PC段不报错，移动端报undefined）。换成Wings后变成了上述错误。我尝试过关闭PJAX、删除所有插件、清空缓存等，都没有效果。 但有一个方法我没有尝试过，就是仅把站点地址改成https（在部署了SSL的前提下），这个改动修复了一些其他插件的问题（包括块编辑器不停掉线无法更新，一些插件无法安装license等），这是我在更换主题之后才发现的。 不知道上述信息对你有没有帮助，如果必要的话我想我可以再建一个站试试。


我这里也出现过，在评论中出现的，未登录用户每次首次评论必现，php8, wp6.1.1 ，看请求返回的错误信息为
```
<b>Warning</b>:  Attempt to read property "comment_author_IP" on null in <b>/www/wwwroot/xxxx/wp-content/plugins/esay-location/easy-location.php</b> on line <b>271</b><br />
```
我查了一下原因，大概是php版本引起的，可以参考这个https://www.weisay.com/blog/wp-themes-php-notice-warning.html
我改了`easy-location.php`第271行代码
把`if ($comment->comment_author_IP &&` 改为 `if (isset($comment->comment_author_IP) && `现在不会报错了

**Reply by ERROR403XD** [reply]
> 留个url？

哦对，我再报一个bug，是Adams有的，Wings好像也有
文章中插入图片会自动加一个wp-image的属性，这个属性会导致图片无法正常显示，头图也受这个影响。
为了插图片我在function里加了一段自动移除这个属性才行（都是GPT老师教的，我不会php）
